File: unix/boot/spp/rpp/rppfor/sdupl.f
      integer function sdupl (str)
      integer str (100)
      integer mem( 60000)
      common/cdsmem/mem
      integer i
      integer length
      integer j
      integer dsget
      j = dsget (length (str) + 1)
      sdupl = j
      i = 1
23000 if (.not.(str (i) .ne. -2))goto 23002
      mem (j) = str (i)
      j = j + 1
23001 i = i + 1
      goto 23000
23002 continue
      mem (j) = -2
      return
      end

File: unix/boot/spp/rpp/rpprat/sdupl.r
#-h-  sdupl			  374  local   12/01/80  15:55:03
# sdupl --- duplicate a string in dynamic storage space
   include  defs

   pointer function sdupl (str)
   character str (ARB)

   DS_COMMON

   integer i
   integer length

   pointer j
   pointer dsget

   j = dsget (length (str) + 1)
   sdupl = j
   for (i = 1; str (i) != EOS; i = i + 1) {
      mem (j) = str (i)
      j = j + 1
      }
   mem (j) = EOS

   return
   end
